### PR TITLE
Dfarr/bugfix  site search error and refactor

### PR DESCRIPTION
**Pull request by DClarkFarr**
Fixing these errors that are preventing the search list from auto populating with suggestions.

Also...refactored, because...i had to. Comments in code.

Screenshot 2024-07-26 at 2 01 46 PM

Notice the: `-1` that is causing the `[i].route` of undefined error.
Screenshot 2024-07-26 at 2 01 53 PM

### CODE REVIEW

File: src/components/SiteSearch.vue [hunk 1]
```diff
@@ -124,14 +122,17 @@ export default class SiteSearch extends Vue {
...
   get suggestedLinks() {
-    return this.quickLinks.filter(i => {
-      let findResult: any = this.searchData.find(
-        data => data.name === i.item.name
+    return this.quickLinks.reduce((acc, i) => {
+      let jsonSearchIndex = this.jsonSearch.findIndex(
+        (data) => data.name === i.item.name
       );
-      let suggestResult: any = this.searchData.indexOf(findResult);
-      this.quickLinkLoc.push(suggestResult);
```
Comment (DClarkFarr) [inline src/components/SiteSearch.vue:132]: Mutating inside of a computed method is super side-effecty. Also, if this ever did re-compute, wouldn't the list just grow and grow indefinitely? 

The whole point was to map suggestedLink -> jsonSearchIndex.  So just add that key to the suggestedLinks.

File: src/components/SiteSearch.vue [hunk 2]
```diff
@@ -103,13 +103,11 @@ export default class SiteSearch extends Vue {
   private resultLimit: Number = 7;
   private fuse: any = null;
   private value: String = "";
-  private quickLinkLoc: any = [];
   private keyEvents: any = [];
   private currentResult: number = 0;
 
   @Prop()
   jsonSearch!: any;
-  searchData = this.jsonSearch;
```
Comment (DClarkFarr) [inline src/components/SiteSearch.vue:112]: Basically...let's copy jsonSearch. Do nothing to mutate the data. And then watch the copied data that never changes for changes. Why not just watch jsonSearch instead?

Action: replace searchData with jsonSearch everywhere

File: src/components/SiteSearch.vue (hunk 1 above)
Comment (DClarkFarr) [reply src/components/SiteSearch.vue:132]: Action: replace quickLinkLoc with suggested.jsonSearchIndex everywhere.

Action: For `limitedResult`, there's a `.refIndex` that points to the jsonSearch object. So use that.

File: src/components/SiteSearch.vue [hunk 3]
```diff
@@ -124,14 +122,17 @@ export default class SiteSearch extends Vue {
...
-    return this.quickLinks.filter(i => {
-      let findResult: any = this.searchData.find(
-        data => data.name === i.item.name
+    return this.quickLinks.reduce((acc, i) => {
+      let jsonSearchIndex = this.jsonSearch.findIndex(
+        (data) => data.name === i.item.name
       );
-      let suggestResult: any = this.searchData.indexOf(findResult);
-      this.quickLinkLoc.push(suggestResult);
-      return suggestResult;
```
Comment (DClarkFarr) [inline src/components/SiteSearch.vue:133]: The real problem here is that `suggestedResult` can be `-1`, which is where the error was coming from

File: src/components/SiteSearch.vue (hunk 2 above)
Comment (joshua-larks) [reply src/components/SiteSearch.vue:112]: I was working on migrating beaker to Vue 2.7 and as I went through this file, this was my concern as well.

File: src/components/SiteSearch.vue (hunk 1 above)
Comment (joshua-larks) [reply src/components/SiteSearch.vue:132]: Didn't notice this on my migration. Nice find!

**Review by joshua-larks — APPROVED** [review]

**Pull request merged**